iceoryx_posh/include/iceoryx_posh/internal/popo/smart_chunk.inl

```diff
@@ -129,7 +129,9 @@ inline const R& SmartChunk<TransmissionInterface, T, H>::getUserHeader() const n
 template <typename TransmissionInterface, typename T, typename H>
 inline T* SmartChunk<TransmissionInterface, T, H>::release() noexcept
 {
-    return m_members.smartChunkUniquePtr.release();
+    auto ptr = cxx::unique_ptr<T>::release(std::move(*m_members.smartChunkUniquePtr));
+    m_members.smartChunkUniquePtr.reset();
```

Comment:
How about overloading `unique_ptr::release` for `optional<unique_ptr<T>>&`? Then this would just become
```cpp
auto ptr = cxx::unique_ptr<T>::release(m_members.smartChunkUniquePtr);
```
The the move and reset would happen within the release function.

Reply:
I wouldn't do this. This is a corner case we currently only have for the smart chunk. I would keep it as simplistic as possible. Otherwise we have to do a `nullopt` check in release and what kind of pointer do you return when one puts in a `nullopt` - or call we the error handler which would introduce nice layers of error handling.

Otherwise the `nullptr` could be returned but then a user may easily misuse this api. They think oh the ptr are always valid therefore release always returns a valid non-null pointer and boom they are proved wrong.

On the other hand a call to the error handler will be also surprising.

Reply:
Well, release could return a `optional<T*>`